arkouda/dataframe.py

```diff
@@ -278,8 +294,19 @@ def __init__(self, initialdata=None, index=None):
             # Initial data is a list of arkouda arrays
             elif isinstance(initialdata, list):
                 # Create string IDs for the columns
-                keys = [str(x) for x in range(len(initialdata))]
+                keys = []
+                if columns is not None:
+                    if any([not isinstance(label, str) for label in columns]):
```

Comment:
Small suggestion, we could pass the generator into `any` directly and avoid the list creation
```suggestion
                    if any(not isinstance(label, str) for label in columns):
```